vyper/builtin_functions/functions.py

```diff
@@ -290,6 +290,32 @@ def build_LLL(self, expr, args, kwargs, context):
             ]
             return LLLnode.from_list(node, typ=ByteArrayType(length.value), location="memory")
 
+        # special handling for slice(<address>.code, start, length)
```

Comment:
It might be nice to refactor the common functionality for `slice(msg.data)` and `slice(<address>.code)` into a helper function

Reply:
I thought about it, but I cannot find a way to extract common functionality between these two since `<address>.code` case needs to handle two cases for `codecopy` and `extcodecopy`, and also, for example, `extcodecopy` has different number of arguments compared to `calldatacopy` used for `msg.data`.